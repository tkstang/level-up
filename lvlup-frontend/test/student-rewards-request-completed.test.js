import React from 'react';
import ReactDOM from 'react-dom';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import thunk from 'redux-thunk';
import configureMockStore from 'redux-mock-store';
import StudentRewardRequestCompleted from '../src/components/Student/requests/request-completed';
import lvlupApp from '../src/reducers/index';

const middlewares = [thunk];
const mockStore = configureMockStore(middlewares);

test('StudentRewardRequestCompleted should render', () => {
  const store = mockStore({ lvlupApp });
  const req = shallow(
    <StudentRewardRequestCompleted reward={{ name: 'Pizza Party' }} requestedReward={{ data: { notes: 'No anchovies' } }} store={store} />,
  );
  expect(shallowToJson(req)).toMatchSnapshot();
});
