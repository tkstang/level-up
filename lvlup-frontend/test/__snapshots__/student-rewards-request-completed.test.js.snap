exports[`test StudentRewardRequestCompleted should render 1`] = `
<div
  className="lvl-table">
  <Table
    as="table"
    celled={true}
    color="orange"
    selectable={true}>
    <TableHeader
      as="thead">
      <TableRow
        as="tr"
        cellAs="td">
        <TableHeaderCell
          as="th"
          className="completed-thead"
          textAlign="center">
          Reward Request Successful!
        </TableHeaderCell>
      </TableRow>
    </TableHeader>
    <TableBody
      as="tbody">
      <TableRow
        as="tr"
        cellAs="td">
        <TableCell
          as="td">
          <Label
            ribbon={true}>
            Requested Reward
          </Label>
          Pizza Party
        </TableCell>
      </TableRow>
      <TableRow
        as="tr"
        cellAs="td">
        <TableCell
          as="td">
          <Label
            ribbon={true}>
            Request Notes
          </Label>
          No anchovies
        </TableCell>
      </TableRow>
    </TableBody>
  </Table>
  <Link
    replace={false}
    to="/student/rewards">
    <Button
      as="button"
      basic={true}
      color="orange">
      Back to Rewards
    </Button>
  </Link>
</div>
`;
